r-pkg/tests/testthat/test-integration.R

```diff
@@ -116,20 +116,20 @@ futile.logger::flog.threshold(0)
         }, regexp = "Query is syntactically valid but 0 documents were matched")
         expect_null(outDT)
     })
-    
+
     # aggregation request
     test_that("es_search works as expected for a simple aggregation request", {
         testthat::skip_on_cran()
-        
+
         outDT <- es_search(
             es_host = "http://127.0.0.1:9200"
             , es_index = "shakespeare"
             , max_hits = 100
             , query = '{"aggs": {"thing": {"terms": {"field": "speaker", "size": 12}}}}'
         )
-        
+
         expect_true(data.table::is.data.table(outDT))
-        expect_true(nrow(outDT) == 4)
+        expect_true(nrow(outDT) == 3)
```

Comment:
Looks like this is failing checks?

Reply:
Ha yeah I kind of thought it might! This is a weird inconsistency between ES <7 and Es 7. I _think_ that what's happening is that previously you were able to have multiple document types in one index and that + something weird in the way we write test data was causing duplicate entries for some indices. I'll have to figure that out to get this merged.

Reply:
nope I was totally wrong. It's a different thing. In the ES7.x one, I'm using the `keyword` type (first introduced in ES6.x) for field `speaker`. That tells Elasticsearch "don't pass the values through a tokenizer, treat the full text as a single level of a categorical".

For ES6 I'm using the `text` type and for ES5 I'm using the `string` type. Those default to breaking down their inputs into tokens with a whitespace tokenizer.

So basically for the ES7 test it thinks there are three unique speakers: `henry iv`, `king`, and `westmoreland`. But for all earlier versions, the same `terms` agg gives you four levels:

```
          thing doc_count
1:        henry        34
2:           iv        34
3:         king        34
4: westmoreland        13
```

I chose the `keyword` type for this field in the ES7 mapping because Elasticsearch removed the use of `fielddata = true` to say "make it possible to do a terms agg".

For now, I'm going to go with the approach of added an explicit check on the version around this test. It's gross so I'll open a "come back and make this less gross" bug, but I feel like it's a thing that will:

1. Give us confidence that this PR doesn't break backwards compatibility of our library with all earlier Elasticsearch versions
2. Give us confidence that `uptasticsearch` can process the result of a `terms` agg from ES7.x correctly